lib/src/services/jira_api.test.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable functional/prefer-immutable-types */
+import * as E from "fp-ts/lib/Either";
+import JiraApi from "jira-client";
+import { JiraClient, jiraClient } from "./jira_api";
+import { boardReturn, searchJiraReturn } from "./test_data/jira_api_data";
+import { readonlyDate } from "readonly-types";
+import { pipe } from "fp-ts/lib/function";
+
+jest.mock("jira-client", () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      searchJira: jest
+        .fn()
+        .mockImplementation(() =>
+          Promise.resolve({ issues: searchJiraReturn })
+        ),
+      getAllBoards: jest
+        .fn()
+        .mockImplementation(() =>
+          Promise.resolve({ values: [{ id: 0, name: "0" }] })
+        ),
+      getConfiguration: jest
+        .fn()
+        .mockImplementation(() => Promise.resolve(boardReturn)),
+      genericGet: jest.fn().mockImplementation((input: string) => {
+        if (input.includes("comment")) return Promise.resolve({ comments: [] });
+        else
+          return input.includes("parent")
+            ? Promise.resolve({ worklogs: [] })
+            : Promise.resolve({
+                worklogs: [
+                  {
+                    author: { name: "Jim" },
+                    started: readonlyDate("2023-05-31T13:46:58.132+1000"),
+                    timeSpentSeconds: 1,
+                    comment: "I did a thing",
+                  },
+                ],
+              });
```

Comment:
By using a `mockResolvedValue` here, you can also avoid duplicating the `Promise.resolve(...)`. 
If you need to use the `mockImplementation`, calculate the return value and then return it in a single `return Promise.resolve(x)`.

Reply:
I'm not sure this can be done as `mockReturnValue(value)` is shorthand for `mockImplementation(() => value);` and as such removes the parameter which is needed for the conditional logic.

Reply:
the alternative is that I can use mockResolvedValueOnce() to mock the returns but I feel like this isn't ideal because if we wanted to extend the number of issues fed into the test case in the future this would not be a very scalable approach.